### PR TITLE
Add WIFI_IMMEDIATE functionality.

### PR DESCRIPTION
**Pull request by cnlohr**

### CODE REVIEW

**Comment by AEFeinstein** [comment]
Ah, this isn't the IGI thing. I know the advantage of immediate packets is speed, but how much risk is there?

**Comment by cnlohr** [comment]
It's that this is only enabled if a given mode chooses to enable it.  (Though I think most modes would do better to use this over queuing up rx packets)

I can put the IGI thing in this.

**Comment by AEFeinstein** [comment]
That's kind of what I was getting at. If you're gonna add an option to
queue or not, I want to be clear on the pros/cons so that we know which to
use when. If one is clearly better than the other, it shouldn't be an
option, and we should just do the better thing. KISS.

I don't mind if the IGI thing is here or in another PR.

**Comment by Brycey92** [comment]
I'm not sure whether my mode will be compatible with interrupt-based tx/rx callbacks, since I track every packet in sequence, including ACKs. Would this cause anything to arrive out-of-order or in the middle of processing another p2p callback?

**Comment by AEFeinstein** [comment]
Can you describe how your mode works? If your logic sends a packet, waits
for the ACK, sends the next packet, etc. it should be fine either way.

**Comment by Brycey92** [comment]
I'm more concerned with the receiving side that waits for an ACK, then waits for a specific packet. As long as TX callbacks are always called before RX callbacks, it should be fine.

**Comment by Brycey92** [comment]
BCP-IP Main Finite State Machine drawio

File: main/modes/mode_flight.c [hunk 1]
```diff
@@ -1459,3 +1554,118 @@ static void flightTimeHighScoreInsert( int insertplace, bool is100percent, char
...
+			m->binencprop = res_codeword;
+			memcpy( m->root, data, sizeof( network_model_t ) - 8 + bones * 3 );
+		}
+	}
+	{
+		int shipCount = ReadUEQ( &assetCounts, 8 );
+		for( i = 0; i < modelCount; i++ )
+		{
+			//XXX XXX XXX
+			peership_t
```
Comment (AEFeinstein) [inline main/modes/mode_flight.c:1639]: dangling?

**Comment by AEFeinstein** [comment]
> It's that this is only enabled if a given mode chooses to enable it. (Though I think most modes would do better to use this over queuing up rx packets)

@cnlohr I still want some guidance as to when a mode should use either method. Other developers are going to be confused by this and not know which to pick.

> I can put the IGI thing in this.

Might as well

File: main/modes/mode_flight.c [hunk 2]
```diff
@@ -1562,18 +1531,96 @@ static void flightTimeHighScoreInsert( int insertplace, bool is100percent, char
...
+		for( ; p != end; p++ )
+		{
+			int ticks = p->validForTicks;
+			if( ticks )
+			{
+				// Render peer/ship.
+
+				int32_t deltaTime = now - p->timeOfUpdate;
+	// Not valid for server.
+	uint32_t timeOfUpdate;  // In our timestamp
```
Comment (AEFeinstein) [inline main/modes/mode_flight.c:1554]: You may want to run this through an autoformatter to at least normalize tabs, spaces, and the like

**Comment by cnlohr** [comment]
My intent was for you to begin testing the changes with IMMEDIATE_MODE + some of the IGI Stuff, while I continued working on the flight mode.  With the changes here, so far, in my apartment, the fighter works much more smoothly.

File: components/hdw-esp-now/espNowUtils.c [hunk 3]
```diff
@@ -26,7 +26,7 @@
 //==============================================================================
 
 #define ESPNOW_CHANNEL 11
-#define WIFI_RATE WIFI_PHY_RATE_MCS7_SGI
+#define WIFI_RATE WIFI_PHY_RATE_MCS6_SGI
```
[the remark this answers is not in the document]
Comment (cnlohr) [reply components/hdw-esp-now/espNowUtils.c:29]: They can 100% interop.  It's just MCS6 seems to work more reliably and further than MCS7 to a notable degree.

[the remark this answers is not in the document]
Comment (cnlohr) [reply components/hdw-esp-now/espNowUtils.c:29]: This has no impact on the radio's receiver hardware.

[the remark this answers is not in the document]
Comment (AEFeinstein) [reply components/hdw-esp-now/espNowUtils.c:29]: Good deal

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**